### PR TITLE
Refactor mixnet, add new type mixnet.Conn.

### PR DESCRIPTION
**Pull request by cjpatton**
RouterContext.AcceptProxy() waits for a proxy to connect to the router
and returns a Conn. ProxyContext.DialRouter() connects to a router and
returns a Conn. Conn implements the net.Conn interface and overloads
Read() and Write() to transmit fixed-length cells.

### CODE REVIEW

File: go/apps/mixnet/conn.go [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+// protocol.
+type Conn struct {
+	net.Conn
+}
+
+// Read a cell from the channel. Return an error if the length
+func (c Conn) Read(msg []byte) (n int, err error) {
+	if n, err = c.Conn.Read(msg); err != nil {
+		return n, err
+	} else if n != CellBytes {
```
Comment (tmroeder) [inline go/apps/mixnet/conn.go]: I think this would be clearer if you pulled out the second and third parts of the if/else sequence:

if n, err = c.Conn.Read(msg); err != nil {
  return n, err
}

if n != CellBytes {
  return n, errCellLength
}

return n, nil


If general, if there's an if (error) { return }, then there usually shouldn't be an else clause; the flow should continue in the main line of the function.

**Comment by tmroeder** [comment]
I only have the one style comment that should be fixed in a few places. Also, I'd like to not have to pick up the merge commits like c619538. Is it possible for you to squash these with a rebase?

File: go/apps/mixnet/conn.go (hunk 1 above)
Comment (cjpatton) [inline go/apps/mixnet/conn.go]: That wouldn't work because n would be out of scope in the last return. I would have to write:
n, err = c.Conn.Read(msg)
if err != nil {
return n, err
} else if n != CellBytes {
return n, errCellLength
}
return n, nil

**Comment by cjpatton** [comment]
No problem!

File: go/apps/mixnet/conn.go (hunk 1 above)
Comment (anonx) [inline go/apps/mixnet/conn.go]: @cjpatton It's not "his style", it is a conventional go code. Here is a related link: https://github.com/golang/go/wiki/CodeReviewComments#indent-error-flow

File: Doc/tpmd.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
+Installing and Using a TPM Emulator
```
Comment (tmroeder) [inline Doc/tpmd.md]: This file got added in 20d7e24, so it doesn't belong in this pull request.

**Comment by tmroeder** [comment]
The file Doc/Go-cloudproxy.pdf was removed in 9d896b6, so it doesn't belong in this pull request.

File: go/apps/mixnet/conn.go [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+type Conn struct {
+	net.Conn
+}
+
+// Read a cell from the channel. Return an error if the length
+func (c Conn) Read(msg []byte) (n int, err error) {
+	n, err = c.Conn.Read(msg)
+	if err != nil {
+		return n, err
+	} else if n != CellBytes {
```
Comment (tmroeder) [inline go/apps/mixnet/conn.go]: Thanks for the change. I'd like to get one more degree of de-indentation:

n, err = c.Conn.Read(msg)
if err != nil {
  return n, err
}

if n != CellBytes {
  return n, errCellLength
}

return n, nil

File: go/apps/mixnet/conn.go [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+var errCellLength error = errors.New("incorrect cell length")
+
+// Conn implements the net.Conn interface. The read and write operations are
+// overloaded to check that only cells are sent between agents in the mixnet
+// protocol.
+type Conn struct {
+	net.Conn
+}
+
+// Read a cell from the channel. Return an error if the length
```
Comment (tmroeder) [inline go/apps/mixnet/conn.go]: Please complete the comment
